### ISSUE TITLE
Rspamd freezes for a long time - Dns timeout not working (libevent 1.4 only)

### ISSUE THREAD

**Issue by AlexeySa** [vstakhov/rspamd#135]
rspamd 0.8.1, Centos 6.6 x64

Example:
8.8.8.8 - google working DNS.
83.242.140.22 - just does not working dns, fictional IP.

Configuration №1 - Google dns - it's OK, rspamd not freezes:
```
options.conf:
    dns {
        timeout = 1s;
        sockets = 16;
        retransmits = 2;
	nameserver = [ "8.8.8.8:50" ];
    }
```

Configuration №2 - not working dns. Rspamd not freezes witch check mail - it's OK:
```
options.conf:
    dns {
        timeout = 1s;
        sockets = 16;
        retransmits = 1;
	nameserver = [ "83.242.140.22" ];
    }
```

Configuration №3 - my rspamd very very long freeze, over 1 hour, bug :(
```
options.conf:
    dns {
        timeout = 1s;
        sockets = 16;
        retransmits = 2;
	nameserver = [ "83.242.140.22" ];
    }
```

Example check cmd:
<code>
cat /tmp/simpleletter | /usr/local/rspamd/bin/rspamc -h 127.0.0.1:11333 -i 84.47.1.1 -F test@mail.ru --helo mail.ru --hostname mail.ru --rcpt test2@mail.com -t 10000</code>

Сonclusion:
If set "retransmits" > 1 rspamd ignore timeout, very long freeze, spam check deferred by timeout :(

**Comment by vstakhov** [comment]
I cannot really reproduce your problem: I tried some fake server with `retransmits = 8` and it worked as expected.

**Comment by AlexeySa** [comment]
This problem is in Centos 6.6 with libevent 1.4.13.

Problem solved after installation libevent 2.0.22 to /usr/local and reconfigure and recompile rspamd.
